Release 5.1.2

DanF-ApexSystems opened the pull request:
To release alongside Restricted v3.1.2

### Added
- Added: Documentation about IL2CPP Setup to prevent code stripping
- Added: Add QueryEntitlement and QueryOwnership tests to the Store demo

### Changed
- Change: Updated Android Documentation
- Change: Packages no longer include .gitattributes files
- Change: Improved UIHighFrequencyPeer2PeerMenu input validation

### Fixed
- Fix: NullReferenceException when using the Copy Log button in the Achievements sample scene
- Fix: Linux build support
- Fix: Updating EOS Config in the UI now saves it to the .json
- Fix: Support for MacOS

JessTello committed:
Fix environment_setup_android.md

DanF-ApexSystems committed:
Update version to 5.1.1

DanF-ApexSystems committed:
Fix for NullReferenceException for Copy Log button in Achievements scene

DanF-ApexSystems committed:
Update version to 5.1.2

DanF-ApexSystems committed:
Convert UIHighFrequencyPeer2PeerMenu.cs to use String.IsNullOrEmpty()

JessTello committed:
Add QueryEntitlement and QueryOwnership tests to the Store demo

A user can QueryEntitlement from the Store demo scene and see a list of entitlements.
A user can QueryOwnership from the Store demo scene and see a list of durable products that are owned.

DanF-ApexSystems committed:
Android documentation reformatting

DanF-ApexSystems committed:
Android documentation update Unity supported versions

DanF-ApexSystems committed:
Android documentation additional reformatting

billfr-ApexSystems committed:
Scripts and prefab updates to display Plugin, EOSSDK, and Native Lib versions in all Sample Apps

Added  the following scripts: EOSSDKVersionInfo.cs, UINativeLibVersion.cs, PluginVersionInterface.cs.  Added new setting in EOSPackageInfo.cs - NativeLibVersion.  Modified UIDemoTemplate.prefab to display Plugin, EOSSDK, and NativeLib versions.

billfr-ApexSystems committed:
Fix for Support ticket 21503423 build fails on Linux

Add support for elif for Linux DLLHBinaryName in SystemDynamicLibrary.cs per Unity Priority support request

DanF-ApexSystems committed:
Ignore .gitattributes files in package export

DanF-ApexSystems committed:
Add Documentation about IL2CPP Setup to prevent code stripping

DanF-ApexSystems committed:
Restore original EOS SDK version of Bindings.cs

billfr-ApexSystems committed:
Version reporting file updates

Moved files into corresponding folders to align with restricted version

billfr-ApexSystems committed:
Updated formatting of version files

Removed PEW headers and added appropriate namespaces for sample files

DanF-ApexSystems committed:
Fix for macOS support

DanF-ApexSystems committed:
Build Windows Native Libraries

DanF-ApexSystems committed:
Update UIHighFrequencyPeer2PeerMenu.cs to validate input

- Added TryParse block
- Checks for negative values

billfr-ApexSystems committed:
Added config.yml to direct users to Epic support

Added file create by Matt Clarke

billfr-ApexSystems committed:
Merge branch 'release-5.1.2' of https://github.com/EOS-Contrib/eos_plugin_for_unity into release-5.1.2

JessTello committed:
Update libDynamicLibraryLoaderHelper.so

JessTello committed:
Fix: EOS UI configuration persisting to JSON

DanF-ApexSystems committed:
Update changelog

billfr-ApexSystems reviewed (changes requested):
Why are changes being made to the EOSSDK files?

billfr-ApexSystems reviewed (changes requested):
Would you please update code comments in code to English?

billfr-ApexSystems reviewed (changes requested):
Remove Apex Header in PluginVersionInterface.cs

billfr-ApexSystems reviewed (changes requested):
Break this commit into specific ones

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Common/UINativeLibVersion.cs:
```diff
@@ -0,0 +1,14 @@
+        private void Start()
+        {
+            var textComp = GetComponent<Text>();
+            textComp.text = $"v-{EOSPackageInfo.NativeLibVersion}";
```
Minor: Add null check
```suggestion
        if (textComp != null)
        {
            textComp.text = $"v-{EOSPackageInfo.NativeLibVersion}";
        }
```

DanF-ApexSystems replied:
Added null check

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Common/UIUnityVersion.cs:
```diff
@@ -0,0 +1,14 @@
+        private void Start()
+        {
+            var textComp = GetComponent<Text>();
+            textComp.text = $"v-{Application.unityVersion}";
```
Minor: Add null check
```suggestion
        if (textComp != null)
        {
            textComp.text = $"v-{Application.unityVersion}";
        }
```

DanF-ApexSystems replied:
Added null check

matt-clarke commented on com.playeveryware.eos/Runtime/Core/SystemDynamicLibrary.cs:
```diff
@@ -102,6 +102,8 @@ public partial class SystemDynamicLibrary
         "DynamicLibraryLoaderHelper_Android";
 #elif UNITY_STANDALONE_OSX
         "libDynamicLibraryLoaderHelper";
+#elif UNITY_STANDALONE_LINUX
+        "libDynamicLibraryLoaderHelper";
```
Minor: Reduce repeated code
```suggestion
#elif UNITY_STANDALONE_OSX || UNITY_STANDALONE_LINUX
        "libDynamicLibraryLoaderHelper";
```

DanF-ApexSystems replied:
Implemented suggested change

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -265,23 +265,60 @@ protected override void ShowInternal()
+                {
+                    Debug.Log("Invalid value: " + mb);
+                    return;
+                }
```
Minor: Needs cleaning up, see above

DanF-ApexSystems replied:
Implemented local variable version, removed else wrapper, and removed final return statement in SetPacketSize to mirror the changes to SetRefreshRate

matt-clarke replied:
The value of `packetSizeMB` isn't assigned to `Peer2PeerManager.packetSizeMB` at all

DanF-ApexSystems replied:
Fixed

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -265,23 +265,60 @@ protected override void ShowInternal()
+            }
+            else
+            {
+                bool attemptParse = int.TryParse(hz, out Peer2PeerManager.refreshRate);
```
Sorry, there's also a subtle bug here that I missed before. It's probably best fixed in a following release anyway, as it's just the samples.

Using `Peer2PeerManager.refreshRate` as the `out` param means the value is always updated.
- If `-5` was passed then the value is update still, despite the logging below saying it wasn't
- If an invalid string is passed in then the function will assign the default value (0) to it

I'd recommend the following, using a local variable for the out param:
```csharp
bool attemptParse = int.TryParse(hz, out int refreshRate);
if (attemptParse)
{
	if (refreshRate < 0)
	{
		Debug.Log("Invalid value (negative): " + hz);
		return;
	}

    Peer2PeerManager.refreshRate = refreshRate;
	Debug.Log("Updated refresh rate to " + refreshRate + " Hz.");
}
else
{
	Debug.Log("Invalid value: " + hz);
	return;
}
```

DanF-ApexSystems replied:
Implemented local variable version

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -265,23 +265,60 @@ protected override void ShowInternal()
+                Debug.Log("Invalid value: empty or null input.");
+                return;
+            }
+            else
```
Minor: This else block is unnecessary because of the `return` above it

DanF-ApexSystems replied:
Removed else wrapper

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -265,23 +265,60 @@ protected override void ShowInternal()
+                else
+                {
+                    Debug.Log("Invalid value: " + hz);
+                    return;
```
Minor: Unnecessary return

DanF-ApexSystems replied:
Removed return

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Peer2Peer/HighFrequency/UIHighFrequencyPeer2PeerMenu.cs:
```diff
@@ -265,23 +265,60 @@ protected override void ShowInternal()
+            }
+            else
+            {
+                bool attemptParse = float.TryParse(mb, out Peer2PeerManager.packetSizeMB);
```
Same applies here about using a local var as the out param

DanF-ApexSystems replied:
Implemented local variable version

matt-clarke commented on Assets/Plugins/Source/Editor/ConfigEditors/ConfigEditor.cs:
```diff
@@ -182,6 +182,10 @@ public void Load()
+#if UNITY_EDITOR
+            AssetDatabase.SaveAssets();
+            AssetDatabase.Refresh();
+#endif
```
Minor: Unnecessary #if check, this is all Editor code
```suggestion
            AssetDatabase.SaveAssets();
            AssetDatabase.Refresh();
```

DanF-ApexSystems replied:
Removed #if check

matt-clarke commented on Assets/Plugins/Source/Editor/EditorWindows/EOSSettingsWindow.cs:
```diff
@@ -256,13 +256,17 @@ private void Save()
+#if UNITY_EDITOR
+            AssetDatabase.SaveAssets();
+            AssetDatabase.Refresh();
+#endif
```
Minor: Unnecessary #if check, this is all Editor code
```suggestion
            AssetDatabase.SaveAssets();
            AssetDatabase.Refresh();
```

DanF-ApexSystems replied:
Removed #if check

matt-clarke commented on com.playeveryware.eos/Runtime/Core/Config/ProductConfig.cs:
```diff
@@ -251,6 +251,11 @@ private void UpdatePlatformConfigDeployments()
+#if UNITY_EDITOR
+                UnityEditor.AssetDatabase.SaveAssets();
+                UnityEditor.AssetDatabase.Refresh();
+#endif
```
Minor: Move this to the base class, to reduce repeated code and the chance of it not being called. 
- Move it to the `Config.OnWriteCompleted()` function
- Update the override of that function in this class to call `base.OnWriteCompleted()`, at the top of the function

DanF-ApexSystems replied:
Moved and updated

matt-clarke commented on com.playeveryware.eos/Runtime/Core/Config/ProductConfig.cs:
```diff
@@ -314,6 +319,11 @@ private void UpdatePlatformConfigClientCredentials()
+#if UNITY_EDITOR
+                UnityEditor.AssetDatabase.SaveAssets();
+                UnityEditor.AssetDatabase.Refresh();
+#endif
```
Minor: As above, this should be moved into the base class

DanF-ApexSystems replied:
Moved

matt-clarke commented on com.playeveryware.eos/Runtime/Core/Config/ProductConfig.cs:
```diff
@@ -331,6 +341,9 @@ protected override void OnWriteCompleted()
             UpdatePlatformConfigClientCredentials();
+#if UNITY_EDITOR
+            UnityEditor.AssetDatabase.Refresh();
+#endif
```
Minor: This looks unnecessary because of the changes above and `OnWriteCompleted()` only being called after `Write()` is called. Regardless, this can definitely be removed after the above suggestions

DanF-ApexSystems replied:
We tested without the .Refresh and found that the data did not update, so we determined we need to keep this line.

matt-clarke replied:
I assume this code wasn't moved into the base class when testing that?
Both this code and the class in [ConfigEditor.cs](https://github.com/EOS-Contrib/eos_plugin_for_unity/pull/1180/files#diff-6b8133ae0ec9698548078ddfe4a27c4f1fc84c01347027607cd19e9bad7fe55d) share the same base class.
Moving this into the function in the base class means that both the code here and in ConfigEditor.cs can be removed

DanF-ApexSystems replied:
Moved to from here and ConfigEditor to Config.cs

matt-clarke commented on com.playeveryware.eos/Runtime/Core/Utility/PluginVersionInterface.cs:
```diff
@@ -0,0 +1,50 @@
+        /// <summary>
+        /// The Hotfix version number
+        /// </summary>
+        public const int HOTFIX = 2;
```
Very Minor: Move this below the `PATCH` variable, it's then easier to parse the exact version by eye

DanF-ApexSystems replied:
Moved this block to below the PATCH block

matt-clarke commented on com.playeveryware.eos/Runtime/Core/EOSPackageInfo.cs:
```diff
@@ -33,7 +33,8 @@ public static class EOSPackageInfo
       
-        public const string Version = "5.1.0";
+        public const string Version = "5.1.2";
+        public const string NativeLibVersion = "5.0.0";
```
Can you explain the need for separate versions for these? I would've expected the native libs to always be rebuilt.
Looking at the included files too they have changed in this PR, but the versions here are different

DanF-ApexSystems replied:
Placeholder to be used for future native lib identifier. We can remove for now until all native lib projects have it implemented.

If we remove, the source code will have to be updated as well.

matt-clarke replied:
Thanks, is it a reliable value right now? If not can you change it to make that obvious (e.g. change it to `todo`)

DanF-ApexSystems replied:
Changed value to "todo".

matt-clarke replied:
The value is still showing as 5.0.0? Approving the review anyway because it's a small detail

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Store/UIStoreMenu.cs:
```diff
@@ -69,7 +81,40 @@ protected override void Update()
                 {
                     catalogueItem1.text = string.Format("{0}, ${1}", CatalogOffers[1].TitleText, StoreManager.GetCurrentPriceAsString(CatalogOffers[1]));
```
This is an existing issue but this should call `base.OnDestroy()` at the bottom of the function

DanF-ApexSystems replied:
Added base.OnDestroy()

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Store/UIStoreMenu.cs:
```diff
@@ -69,7 +81,40 @@ protected override void Update()
+               
+
             }
+             // Entitlements
```
This code has a ton of formatting problems and inconsistencies.
e.g. indentation, mismatched use of curly braces in some places and in others none, comments/strings in Spanish

DanF-ApexSystems replied:
Did a formatting/indentation pass

matt-clarke commented on Assets/Scripts/StandardSamples/UI/Store/UIStoreMenu.cs:
```diff
@@ -80,9 +125,89 @@ public void OnQueryOffersClick()
+                ? fromOffers.ToArray()
+                : (durableCatalogItemIds ?? System.Array.Empty<string>());
+
+            if (ids.Length == 0) { Debug.LogWarning("[UI] No durable IDs."); return; }
```
Should be over multiple lines, particularly as it has a `return` statement

DanF-ApexSystems replied:
Split into multiple lines

matt-clarke commented on Assets/Scripts/EOSStoreManager.cs:
```diff
@@ -66,13 +76,158 @@ public void OnLoggedOut()
+            }
+        }
+        public void QueryEntitlements(bool includeRedeemed = true)
+        { var uid = EOSManager.Instance.GetLocalUserId();
```
Again, more formatting problems. Many more across the file

DanF-ApexSystems replied:
Did a formatting/indentation pass

matt-clarke commented on Assets/Scripts/EOSStoreManager.cs:
```diff
@@ -66,13 +76,158 @@ public void OnLoggedOut()
+            var options = new QueryEntitlementsOptions
+            {
+                LocalUserId = EOSManager.Instance.GetLocalUserId(),
+                EntitlementNames = null, // null 
```
Pointless comment

DanF-ApexSystems replied:
Removed comment

matt-clarke commented on Assets/Scripts/EOSStoreManager.cs:
```diff
@@ -66,13 +76,158 @@ public void OnLoggedOut()
+                    Debug.Log($"  - [{i}] Name={entitlement.Value.EntitlementName}, ItemId={entitlement.Value.CatalogItemId}, Redeemed={entitlement.Value.Redeemed}");
+                    _entitlements.Add(entitlement.Value);
+                }
+                else Debug.LogWarning($"  - CopyEntitlementByIndex FAILED at index {i}: {res}");
```
Add curly braces

DanF-ApexSystems replied:
Added curly braces

matt-clarke commented on Assets/Scripts/EOSStoreManager.cs:
```diff
@@ -66,13 +76,158 @@ public void OnLoggedOut()
+            {
+                LocalUserId = EOSManager.Instance.GetLocalUserId(),
+                CatalogItemIds = utf8Ids
+                // CatalogNamespace = null // 
```
Commented out code

matt-clarke commented on Assets/Scripts/EOSStoreManager.cs:
```diff
@@ -147,6 +303,13 @@ public void CheckOutOverlay(int index)
             Debug.Log($"Checkout {checkoutCallbackInfo.ResultCode}");
+            if (checkoutCallbackInfo.ResultCode == Epic.OnlineServices.Result.Success)
+            {
+                // Refrescar ambas vistas para el AC
```
Comments in Spanish

DanF-ApexSystems replied:
Replaced with English comment

matt-clarke commented on Assets/Scripts/EOSStoreManager.cs:
```diff
@@ -176,5 +339,34 @@ public string GetOriginalPriceAsString(CatalogOffer? catalogOffer)
+            Debug.Log($"[StoreManager] RedeemEntitlements → {info.ResultCode}");
+
+           
+            QueryEntitlements(includeRedeemed: true);
```
Incorrect indent

DanF-ApexSystems replied:
Corrected indent

DanF-ApexSystems committed:
Remove unnecessary if block in ConfigEditor.cs

DanF-ApexSystems committed:
Remove unnecessary if block in EOSSettingWindow.cs

DanF-ApexSystems committed:
Add null check to UINativeLibVersion.cs

DanF-ApexSystems committed:
UIHighFrequencyPeer2PeerMenu.cs cleanup

- Removed unnecessary else blocks
- Removed unnecessary returns
- Changed Peer2PeerManager.refreshRat to a local variable
- Changed Peer2PeerManager.packetSizeMB to a local variable

DanF-ApexSystems committed:
Add null check to UIUnityVersion.cs

DanF-ApexSystems committed:
Reduce repeated code in SystemDynamicLibrary.cs

DanF-ApexSystems committed:
Define the version number components in the same order they are combined

DanF-ApexSystems committed:
Remove Apex Systems header from PluginVersionInterface.cs

DanF-ApexSystems committed:
Add comment to Bindings.cs identifying the temporary block of code

JessTello committed:
Refactor: formatting corrections and structural cleanup

DanF-ApexSystems committed:
Fix comment spelling

DanF-ApexSystems committed:
Call base.OnWriteCompleted() from the override class

matt-clarke commented on Assets/Plugins/Source/Editor/ConfigEditors/ConfigEditor.cs:
```diff
@@ -182,6 +182,8 @@ public void Load()
         {
             config.Write(prettyPrint);
+            AssetDatabase.SaveAssets();
+            AssetDatabase.Refresh();
```
See also [this comment](https://github.com/EOS-Contrib/eos_plugin_for_unity/pull/1180/files#r2545968364), which suggests moving this into the base class instead

DanF-ApexSystems replied:
Moved to Config.cs

DanF-ApexSystems committed:
Change SessionsAndMatchmaking Selected Button On Close to LogoutButton

billfr-ApexSystems committed:
Update EOSPackageInfo.cs

Corrected NativeLibSDKVersion value to "todo value"

JessTello committed:
Update UIHighFrequencyPeer2PeerMenu.cs

JessTello committed:
Refactor ConfigEditor

Moving function into base class

JessTello committed:
Update ConfigEditor.cs
